presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -257,4 +259,21 @@ public Session getSession()
             return session;
         }
     }
+
+    private static class Result
+            implements Rule.Result
+    {
+        private Optional<PlanNode> transformed = Optional.empty();
+
+        @Override
+        public Rule.Result transformTo(PlanNode planNode)
+        {
+            return this;
+        }
+
+        private Optional<PlanNode> getTransformed()
```

Comment:
this can be public (since this is nested class) or package private. Private might suggest it should be used by wrapping class too.